lms/djangoapps/grades/models.py

```diff
@@ -575,6 +575,24 @@ def prefetch(cls, course_id, users):
             cls.objects.filter(user_id__in=[user.id for user in users], course_id=course_id)
         }
 
+    @classmethod
+    def prefetch_by_user(cls, user, course_keys):
+        """
+        Prefetches grades for the given user for each given course
+
+        Note that this may clobber data from prefetches done by a different access pattern
+        Interleave different means of prefetching at your peril
+        """
+        cache = get_cache(cls._CACHE_NAMESPACE)
+        grades = cls.objects.filter(user_id=user.id)
```

Comment:
Maybe just join to enrollments (or subquery by enrollments)? `course_id__in=CourseEnrollment.objects.filter(user=user)`, or the like?

Reply:
Sure yeah, I forget about subqueries because they come from hell but that seems like a good compromise to the api weirdness at play here.